src/librustc_borrowck/diagnostics.rs

```diff
@@ -911,6 +956,50 @@ You can find more information about borrowing in the rust-book:
 http://doc.rust-lang.org/stable/book/references-and-borrowing.html
 "##,
 
+E0508: r##"
+This error occurs when a value is moved out of a non-copy fixed size array.
+
+Example of erroneous code:
+
+```compile_fail
+struct NonCopy;
+
+fn main() {
+    let array = [NonCopy; 1];
+    let _value = array[0]; // error: cannot move out of type `[NonCopy; 1]`,
+                           //        a non-copy fixed-size array
+}
+```
+
+We try to move the first element out of the array, but this is not
```

Comment:
I'd prefer to avoid to use "we". Please replace it with:

"The first element was moved out of the array, but this is not..."

You can eventually add "in the previous example" or something similar if you want to.